components/renku_data_services/notebooks/api/classes/data_service.py

```diff
@@ -4,22 +4,27 @@
 from typing import Any, NamedTuple, Optional, cast
 from urllib.parse import urljoin, urlparse
 
-import requests
+import httpx
 from sanic.log import logger
 
+from renku_data_services.base_models import APIUser
+from renku_data_services.crc.db import ResourcePoolRepository
+from renku_data_services.crc.models import ResourceClass, ResourcePool
+from renku_data_services.notebooks.api.classes.repository import (
+    INTERNAL_GITLAB_PROVIDER,
+    GitProvider,
+    OAuth2Connection,
+    OAuth2Provider,
+)
+from renku_data_services.notebooks.api.schemas.server_options import ServerOptions
 from renku_data_services.notebooks.errors.intermittent import IntermittentError
-from renku_data_services.notebooks.errors.programming import ConfigurationError
 from renku_data_services.notebooks.errors.user import (
     AuthenticationError,
     InvalidCloudStorageConfiguration,
     InvalidComputeResourceError,
     MissingResourceError,
 )
 
-from ..schemas.server_options import ServerOptions
-from .repository import INTERNAL_GITLAB_PROVIDER, GitProvider, OAuth2Connection, OAuth2Provider
-from .user import User
-
 
 class CloudStorageConfig(NamedTuple):
```

Comment:
Just a question: any reason to use a NamedTuple rather than a Dataclass ?

Reply:
This is just what was used in the old notebooks code that we moved over. So I did not touch it. And considering that we want to eventually just fully remove this code I dont want to touch it now.

Reply:
Fair enough